tests/unit/test_agw_network_configurator.py

```diff
@@ -0,0 +1,1037 @@
+    AGWInstallerNetworkConfigurator,
+)
+
+
+class TestAGWInstallerNetworkConfigurator(unittest.TestCase):
+    TEST_NETWORK_INTERFACES = ["eth0", "eth1"]
+
+    def setUp(self) -> None:
+        self.timestamp = time.time()
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR"  # noqa: E501, W505
+    )
+    @patch("builtins.open", new_callable=mock_open())
+    def test_given_agw_installer_network_configurator_when_prepare_interfaces_configuration_dir_if_doesnt_exist_then_configuration_dir_gets_created_properly(  # noqa: E501
+        self, mocked_open_file, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES, None, None
+        )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+        self.assertTrue(os.path.exists(ifaces_dir))
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR"  # noqa: E501, W505
+    )
+    def test_given_agw_installer_network_configurator_when_prepare_interfaces_configuration_dir_if_doesnt_exist_then_interfaces_configuration_dir_is_written_to_etc_network_interfaces_file(  # noqa: E501
+        self, ifaces_dir
+    ):
+        ifaces_dir.return_value = f".tox/.tmp/interfaces.d_{self.timestamp}"
+        with patch("builtins.open", mock_open()) as mocked_open_file:
+            AGWInstallerNetworkConfigurator(
+                self.TEST_NETWORK_INTERFACES, None, None
+            )._prepare_interfaces_configuration_dir_if_doesnt_exist()
+            mocked_open_file.assert_called_with("/etc/network/interfaces", "w")
+            mocked_open_file().write.assert_called_once_with(f"source-directory {ifaces_dir}")
+
+    def test_given_agw_installer_network_configurator_when_static_eth0_address_given_then_correct_eth0_configuration_is_retured(  # noqa: E501
+        self,
+    ):
+        test_static_ip_address = "1.2.3.4/24"
+        test_static_gw_ip_address = "4.3.2.1"
+        expected_config = [
+            "auto eth0",
+            "iface eth0 inet static",
+            "address 1.2.3.4",
+            "netmask 255.255.255.0",
+            "gateway 4.3.2.1",
+        ]
+        agw_network_configurator = AGWInstallerNetworkConfigurator(
+            self.TEST_NETWORK_INTERFACES,
+            test_static_ip_address,
+            test_static_gw_ip_address,
+        )
+        self.assertEqual(agw_network_configurator._eth0_config, expected_config)
+
+    @patch(
+        "magma_access_gateway_installer.agw_network_configurator.AGWInstallerNetworkConfigurator.INTERFACES_DIR",  # noqa: E501
+        ".tox/.tmp",
+    )
+    def test_given_agw_installer_network_configurator_when_static_eth0_address_given_then_correct_message_is_logged_while_creating_interface_config_file(  # noqa: E501
```

Comment:
Do we really want to write unit tests for log writing? It feels a bit overkill

Reply:
It's already done, so I don't see a reason for removing it. Although, we might agree to not write those anymore in the future

Reply:
The reason for removing it would be that we don't want unit tests to fail when you change the log message.

Reply:
Agreed, already discussed in the daily

Reply:
Tests refactored.